packages/plans-grid-next/src/components/comparison-grid/hooks/useCompactViewOrder.ts

```diff
@@ -0,0 +1,87 @@
+import {
+	PlanSlug,
+	URL_FRIENDLY_TERMS_MAPPING,
+	UrlFriendlyTermType,
+	findFirstSimilarPlanKey,
+	isFreePlan,
+} from '@automattic/calypso-products';
+import { WpcomPlansUI } from '@automattic/data-stores';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { useEffect } from 'react';
+import { GridPlan } from '../../../types';
+
+type Props = {
+	/**
+	 * Selected plan interval
+	 */
+	intervalType: UrlFriendlyTermType;
+	/**
+	 * The map of displayed plans for the current interval
+	 */
+	gridPlansIndex: { [ key: string ]: GridPlan };
+};
+
+/**
+ *
+ * When intervalType change trickles down to the ComparisonGrid as a prop (intervalType),
+ * we must detect this prop change and transform the current plan order into a set of plans
+ * that align with the currently selected plan interval without causing glitches or flashes.
+ *
+ * This hook serves as the solution. The hook accomplishes two main tasks:
+ * - Resolves the correct compact view plan order, providing it immediately in the current render cycle to prevent glitches.
+ * - Updates the WpcomPlansUI store with the most up-to-date plan order as required.
+ *
+ * ******
+ * @typedef {Object} Param1
+ * @property {UrlFriendlyTermType} intervalType - Selected plan interval
+ * @property {Object.<string, GridPlan>} gridPlansIndex - The map of displayed plans for the current interval
+ * @returns The list of gridPlans sorted by the accurate compact view order.
+ */
+
+function useCompactViewOrder( { intervalType, gridPlansIndex }: Props ): GridPlan[] {
```

Comment:
It doesn't look like you need to pass `gridPlansIndex` as a prop. Why not just access it from context? There's no potential use of this hook outside the context provider.

Reply:
> add's some complexity

Everything uses this context, and if we end up refactoring down the line we'll do so holistically. I think the complexity comes from reading it elsewhere and passing it along as a prop. that's what context is meant to serve to begin with

> opens it up to unnecessary renders without a clear advantage in readability.

I don't see that. whether read here through context or at the consuming end and passed as a prop is the same thing in terms of referential concerns